plugins/uefi/fu-plugin-uefi.c

```diff
@@ -613,40 +620,120 @@ load_custom_esp (FuPlugin *plugin, GError **error)
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch (pattern, fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
 	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_efivars (FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GPtrArray) files = NULL;
 
+	/* delete any efivars matching the glob */
+	files = fu_common_get_files_recursive ("/sys/firmware/efi/efivars", error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch ("fwupdate-*", fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
+	}
 	return TRUE;
 }
 
 gboolean
-fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	fwup_resource *re;
-	gint supported;
-	g_autoptr(fwup_resource_iter) iter = NULL;
-	g_autofree gchar *name = NULL;
-	const gchar *ux_capsule_str = "Disabled";
 
-	/* supported = 0 : ESRT unspported
-	   supported = 1 : unlocked, ESRT supported
-	   supported = 2 : it is locked but can be unlocked to support ESRT
-	   supported = 3 : it is locked, has been marked to be unlocked on next boot
-			   calling unlock again is OK.
-	 */
-	supported = fwup_supported ();
-	if (supported == 0) {
+	/* get the supported status */
+	data->esrt_status = fwup_supported ();
+	if (data->esrt_status == FWUP_SUPPORTED_STATUS_UNSUPPORTED) {
 		g_set_error_literal (error,
 				     FWUPD_ERROR,
 				     FWUPD_ERROR_NOT_SUPPORTED,
 				     "UEFI firmware updating not supported");
 		return FALSE;
 	}
 
-	if (supported == 2) {
+	/* load any overriden options */
+	if (!fu_plugin_uefi_set_custom_mountpoint (plugin, error))
+		return FALSE;
+
+	/* get the default compiled-in value for the ESP mountpoint */
+#ifdef HAVE_FWUP_GET_ESP_MOUNTPOINT
+	if (data->esp_path == NULL)
+		data->esp_path = g_strdup (fwup_get_esp_mountpoint ());
+#endif
+
+	/* delete any existing .cap files to avoid the small ESP partition
+	 * from running out of space when we've done lots of firmware updates
+	 * -- also if the distro has changed the ESP may be different anyway */
+	if (data->esp_path != NULL) {
+		if (g_file_test ("/sys/firmware/efi/efivars/"
+				 "BootNext-8be4df61-93ca-11d2-aa0d-00e098032b8c",
+				 G_FILE_TEST_EXISTS)) {
+			g_debug ("detected BootNext, not cleaning up");
+		} else {
+			if (!fu_plugin_uefi_delete_old_capsules (plugin, error))
+				return FALSE;
+			if (!fu_plugin_uefi_delete_old_efivars (plugin, error))
+				return FALSE;
+		}
+	}
+
+	/* save in report metadata */
+	if (data->esp_path != NULL) {
+		g_debug ("ESP mountpoint set as %s", data->esp_path);
```

Comment:
No need to log this debug output here too, it's also in `fu_plugin_uefi_set_custom_mountpoint`

Reply:
I'll drop the one in `fu_plugin_uefi_set_custom_mountpoint()` as that will only be shown in the case the user changed the default, and I really wanted "what we're going to try and use" in the debug logs for debugging.